src/tools/dev/scripts/run-build-visit

```diff
@@ -0,0 +1,364 @@
+      mkdir /usr/gapps/visit/thirdparty_shared/$version
+   fi
+   if [ ! -d /usr/gapps/visit/thirdparty_shared/$version/mesagl ]
+   then
+      mkdir /usr/gapps/visit/thirdparty_shared/$version/mesagl
+   fi
+   export CC=/usr/gapps/gcc/gcc-9.1/bin/gcc
+   export CXX=/usr/gapps/gcc/gcc-9.1/bin/g++
+   export LD_LIBRARY_PATH=/usr/gapps/gcc/gcc-9.1/lib64:/usr/gapps/gcc/gcc-9.1/lib
+   export PATH=/usr/gapps/gcc/gcc-9.1/bin:$PATH
+   env CC=/usr/gapps/gcc/gcc-9.1/bin/gcc CXX=/usr/gapps/gcc/gcc-9.1/bin/g++ \
+   ./$build_visit_script --required --optional --parallel --mpich --mesagl --uintah --no-adios2 --no-visit --thirdparty-path /usr/gapps/visit/thirdparty_shared/$version/mesagl --makeflags -j4
+   mv ${build_visit_script}_log ${build_visit_script}_log.mesagl
+fi
+
+#
+# Build using opengl on kickit.
+#
+if [ $kickit_opengl = true ]
+then
+   echo "Building using opengl on kickit"
+   ran_build_visit=true
+   if [ -f ${build_visit_script}_log.opengl ]
+   then
+      mv ${build_visit_script}_log.opengl ${build_visit_script}_log
+   fi
+   if [ ! -d /usr/gapps/visit/thirdparty_shared/$version ]
+   then
+      mkdir /usr/gapps/visit/thirdparty_shared/$version
+   fi
+   if [ ! -d /usr/gapps/visit/thirdparty_shared/$version/opengl ]
+   then
+      mkdir /usr/gapps/visit/thirdparty_shared/$version/opengl
+   fi
+   export CC=/usr/gapps/gcc/gcc-9.1/bin/gcc
+   export CXX=/usr/gapps/gcc/gcc-9.1/bin/g++
+   export LD_LIBRARY_PATH=/usr/gapps/gcc/gcc-9.1/lib64:/usr/gapps/gcc/gcc-9.1/lib
+   export PATH=/usr/gapps/gcc/gcc-9.1/bin:$PATH
+   env CC=/usr/gapps/gcc/gcc-9.1/bin/gcc CXX=/usr/gapps/gcc/gcc-9.1/bin/g++ \
+   ./$build_visit_script --skip-opengl-context-check --required --optional --parallel --mpich --osmesa --uintah --no-adios2 --no-visit --thirdparty-path /usr/gapps/visit/thirdparty_shared/$version/opengl --makeflags -j4
+   mv ${build_visit_script}_log ${build_visit_script}_log.opengl
+fi
+
+#
+# Build on quartz.
+#
+if [ $quartz = true ]
+then
+   echo "Building on quartz"
+   ran_build_visit=true
+   if [ -f ${build_visit_script}_log.quartz ]
+   then
+      mv ${build_visit_script}_log.quartz ${build_visit_script}_log
+   fi
+   if [ ! -d /usr/workspace/wsa/visit/visit/thirdparty_shared/$version ]
+   then
+      mkdir /usr/workspace/wsa/visit/visit/thirdparty_shared/$version
+   fi
+   if [ ! -d /usr/workspace/wsa/visit/visit/thirdparty_shared/$version/toss3 ]
+   then
+      mkdir /usr/workspace/wsa/visit/visit/thirdparty_shared/$version/toss3
+   fi
+   module load gcc/7.3.0
+   env PAR_COMPILER=/usr/tce/packages/mvapich2/mvapich2-2.3-gcc-7.3.0/bin/mpicc \
+       PAR_COMPILER_CXX=/usr/tce/packages/mvapich2/mvapich2-2.3-gcc-7.3.0/bin/mpicxx \
+       PAR_INCLUDE=-I/usr/tce/packages/mvapich2/mvapich2-2.3-gcc-7.3.0/include \
+       ./$build_visit_script --required --optional --mesagl --uintah --parallel --qt510 --no-pyside --no-visit --thirdparty-path /usr/workspace/wsa/visit/visit/thirdparty_shared/$version/toss3 --makeflags -j16
```

Comment:
Isn't --llvm  required to be listed when --mesagl is used ?

Reply:
It gets included with `--optional`.

Reply:
Interesting, I did not know that. Thanks.

Reply:
That might be a good idea. What would be better is if we could get build_visit to correctly handle dependencies like this that aren't first-class options in and of themselves. (eg we don't need LLVM unless we want osmesa or mesagl).